### PR TITLE
feat: c-a views components events migration to segment

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
C-A views component events migration to segment

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/BrowserTab/index.js [hunk 1]
```diff
@@ -1,80 +1,63 @@
-import React, { useState, useRef, useEffect, useCallback } from 'react';
```
Comment (tommasini) [inline app/components/Views/BrowserTab/index.js:1]: It's weird why this import changed, will revert this.

File: app/components/Views/AddAccountActions/AddAccountActions.tsx [hunk 2]
```diff
@@ -17,22 +16,24 @@ import Engine from '../../../core/Engine';
...
   const { navigate } = useNavigation();
+  const { trackEvent } = useMetrics();
   const [isLoading, setIsLoading] = useState(false);
 
   const openImportAccount = useCallback(() => {
     navigate('ImportPrivateKeyView');
     onBack();
-    AnalyticsV2.trackEvent(MetaMetricsEvents.ACCOUNTS_IMPORTED_NEW_ACCOUNT, {});
-  }, [navigate, onBack]);
+    trackEvent(MetaMetricsEvents.ACCOUNTS_IMPORTED_NEW_ACCOUNT, {});
```
Comment (NicolasMassart) [inline app/components/Views/AddAccountActions/AddAccountActions.tsx:29]:
```suggestion
    trackEvent(MetaMetricsEvents.ACCOUNTS_IMPORTED_NEW_ACCOUNT);
```

File: app/components/Views/AddAccountActions/AddAccountActions.tsx [hunk 3]
```diff
@@ -17,22 +16,24 @@ import Engine from '../../../core/Engine';
...
+  }, [navigate, onBack, trackEvent]);
 
   const openConnectHardwareWallet = useCallback(() => {
     navigate(Routes.HW.CONNECT);
     onBack();
-    AnalyticsV2.trackEvent(MetaMetricsEvents.CONNECT_HARDWARE_WALLET, {});
-  }, [onBack, navigate]);
+    trackEvent(MetaMetricsEvents.CONNECT_HARDWARE_WALLET, {});
```
Comment (NicolasMassart) [inline app/components/Views/AddAccountActions/AddAccountActions.tsx:35]:
```suggestion
    trackEvent(MetaMetricsEvents.CONNECT_HARDWARE_WALLET);
```

File: app/components/Views/AddAccountActions/AddAccountActions.tsx [hunk 4]
```diff
@@ -41,15 +42,15 @@ const AddAccountActions = ({ onBack }: AddAccountActionsProps) => {
 
       const { addedAccountAddress } = await KeyringController.addNewAccount();
       PreferencesController.setSelectedAddress(addedAccountAddress);
-      AnalyticsV2.trackEvent(MetaMetricsEvents.ACCOUNTS_ADDED_NEW_ACCOUNT, {});
+      trackEvent(MetaMetricsEvents.ACCOUNTS_ADDED_NEW_ACCOUNT, {});
```
Comment (NicolasMassart) [inline app/components/Views/AddAccountActions/AddAccountActions.tsx:45]:
```suggestion
      trackEvent(MetaMetricsEvents.ACCOUNTS_ADDED_NEW_ACCOUNT);
```

File: app/components/Views/Asset/index.js [hunk 5]
```diff
@@ -475,14 +479,11 @@ class Asset extends PureComponent {
     const onBuy = () => {
       navigation.navigate(Routes.RAMP.BUY);
       InteractionManager.runAfterInteractions(() => {
```
Comment (NicolasMassart) [inline app/components/Views/Asset/index.js:481]: Remove InteractionManager

File: app/components/Views/BrowserTab/index.js [hunk 6]
```diff
@@ -101,7 +98,9 @@ import { TextVariant } from '../../../component-library/components/Texts/Text';
 import { regex } from '../../../../app/util/regex';
 import { selectChainId } from '../../../selectors/networkController';
 import { BrowserViewSelectorsIDs } from '../../../../e2e/selectors/BrowserView.selectors';
-import { trackDappVisitedEvent } from '../../../analytics';
+import { useMetrics } from '../../../components/hooks/useMetrics';
+import trackDappVisitedEvent from '../../../util/metrics/trackDappVisited';
```
Comment (NicolasMassart) [inline app/components/Views/BrowserTab/index.js:102]: Do not update trackDappVisitedEvent in this PR

**Review by NicolasMassart — COMMENTED** [review]
Some issues to fix:
- trackDappVisitedEvent should not be updated here (and it should be moved instead of added)
- no need for empty properties object in trackEvent calls
- missing one InteractionManager removal

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**